projects/plugins/jetpack/modules/masterbar/profile-edit.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * WP-Admin Profile edit.
+ *
+ * @package Jetpack
+ */
+
+use Automattic\Jetpack\Connection\Manager as Connection_Manager;
+
+/**
+ * Hides profile fields for WordPress.com connected users.
+ *
+ * @param WP_User $user The current WP_User object.
+ */
+function masterbar_hide_profile_fields( $user ) {
+	$user_id            = $user->ID;
+	$connection_manager = new Connection_Manager( 'jetpack' );
+	if ( ! $connection_manager->is_user_connected( $user_id ) ) {
+		// If this is a local user, show the default UX.
+		return;
+	}
+	// Since there is no hook for altering profile fields, we will use CSS and JS.
+	$name_info_wpcom_link_message    = __( '<a href="https://wordpress.com/me">WordPress.com users can change their profile\'s basic details ( i.e., First Name, Last Name, Display Name, About ) in WordPress.com Profile settings.</a>', 'jetpack' );
```

Comment:
Nitpick: I think we should make only the text link translatable, without the HTML link:

`$name_info_wpcom_link_message =  '<a href="https://wordpress.com/me">' . __( 'foo', 'bar' ) . '</a>';`

Reply:
Thanks, fixed here 2a22acc